node/trace/agent.js

```diff
@@ -43,7 +44,7 @@ function Agent(options) {
     // incoming reuqests
     self.serviceName = options.serviceName || null;
 
-    self.reporter = options.reporter || self.reporter;
+    self.reporter = options.reporter || nullReporter;
```

Comment:
I'd mildly prefer to keep the default on the prototype, changing this instead to `if (options.reporter) self.reporter = options.reporter;`